octoprint_mrbeam/materials.py

```diff
@@ -26,122 +29,24 @@ def __init__(self, plugin):
-
-    def put_custom_material(self, key, material):
-        """Sanitize and put material. If key exists, material will be
-        overwritten.
+        self.custom_materials = self._load_materials_from_yaml()
 
-        :param key: String unique material key
-        :param material: Dict of material data
-        :return: Boolean success
-        """
-        self._load()
-        res = None
+    def _load_materials_from_yaml(self):
+        custom_materials = {}
 
         try:
-            if dict_get(material, ['laser_type']) == "MrBeamII-1.0":
-                material["laser_model"] = '0'
-                del material["laser_type"]
-            if "model" in material:
-                material["device_model"] = material.pop("model")
-            if "compatible" in material:
-                material.pop("compatible")
-            if "customBeforeElementContent" in material:
-                material.pop("customBeforeElementContent")
-
-            self.custom_materials[key.strip()] = material
-            res = True
-        except:
-            self._logger.exception(
-                "Exception while putting materials: key: %s, data: %s", key, material
-            )
-            res = False
-        if res:
-            res = self._save()
-        return res
+            if os.path.isfile(self.custom_materials_file):
+                with open(self.custom_materials_file) as yaml_file:
+                    content = yaml.safe_load(yaml_file)
+                    custom_materials = content.get("custom_materials", {})
 
-    def delete_custom_material(self, key):
-        """Deletes custom material if existing.
+            self._logger.info("{} custom materials loaded".format(len(custom_materials)))
+        except Exception as e:
+            self._logger.exception("Exception while loading custom materials: {}".format(e))
 
-        :param keys: String or list: key or list of keys to delete
-        :return: Boolean success
-        """
-        self._load()
-        count = 0
-        res = True
-
-        key_list = key
-        if isinstance(key_list, basestring):
-            key_list = [key_list]
-
-        if key_list:
-            try:
-                for k in key_list:
-                    try:
-                        del self.custom_materials[k]
-                        count += 1
-                    except ValueError:
-                        pass
-            except:
-                self._logger.exception(
-                    "Exception while deleting materials: key: %s", key
-                )
-                res = False
-            if res and count > 0:
-                res = self._save()
-        return res
+        return custom_materials
 
-    def reset_all_custom_materials(self):
-        self._logger.info("Resetting all custom material settings!!!!")
-        self.custom_materials = {}
-        self._save(force=True)
-
-    def _load(self, force=False):
-        if not self.custom_materials_loaded or force:
```

Comment:
why aren't we using this property anymore? shall we remove it?

Reply:
You mean self.custom_materials_loaded? Or force?
custom_materials_loaded isn't necessary because we load them in the __init__
and force was to read them from the file again, but in the current implementation it isn't necessary.
Both are removed in the current code